pkg/dbutil/common.go

```diff
@@ -255,3 +280,56 @@ func SetSnapshot(ctx context.Context, db *sql.DB, snapshot string) error {
 	_, err := db.ExecContext(ctx, sql)
 	return errors.Trace(err)
 }
+
+// GetTiDBVersion returns the tidb's version
+func GetTiDBVersion(ctx context.Context, db *sql.DB) (string, error) {
+	/*
+		example in tidb:
+		mysql> SELECT tidb_version()\G
+		*************************** 1. row ***************************
+		tidb_version(): Release Version: v2.1.0-beta-260-gd4e0885
+		Git Commit Hash: d4e08853e4d4d867676f0f17ec6e5b15d5a0bdf8
+		Git Branch: master
+		UTC Build Time: 2018-08-21 03:59:16
+		GoVersion: go version go1.10.3 darwin/amd64
+		Race Enabled: false
+		TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
+		Check Table Before Drop: false
+		1 row in set (0.00 sec)
+	*/
+	query := "SELECT tidb_version()"
+	result, err := db.QueryContext(ctx, query)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	var version sql.NullString
+	for result.Next() {
+		err := result.Scan(&version)
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+		break
+	}
+
+	if version.Valid {
+		// TODO: need analyse the tidb version string
+		return version.String, nil
+	}
+
+	return "", errors.New("can't get the tidb's version")
+}
+
+// IsTiDB returns true if this database is tidb
+func IsTiDB(ctx context.Context, db *sql.DB) (bool, error) {
```

Comment:
I think `select version() ` is better than `select tidb_version()`.

```
mysql> select version();
+--------------------------------------+
| version()                            |
+--------------------------------------+
| 5.7.10-TiDB-v2.1.0-beta-173-g7e48ab1 |
+--------------------------------------+
1 row in set (0.01 sec)
```

Reply:
done